public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -94,6 +111,27 @@ export const GeneratePopoverBody: React.FC<{
     );
   });
 
+  const getInsightParams = (contextObj: ContextObj | undefined) => {
```

Comment:
Feel it will be better that we move this util function out of the render function.

Reply:
Done